query/src/main/java/org/infinispan/query/Indexer.java

```diff
@@ -0,0 +1,38 @@
+package org.infinispan.query;
+
+import java.util.concurrent.CompletionStage;
+
+/**
+ * The indexer is used to interact directly with a cache's index.
+ *
+ * @since 11.0
+ */
+public interface Indexer {
+
+   /**
+    * Deletes all the indexes from a cache and rebuilds them from the existing data.
+    * This operation can take a long time depending on the cache size, and searches
+    * during this process will be affected.
+    */
+   CompletionStage<Void> run();
+
+   /**
+    * Re-indexes only the values associated with the provided keys.
+    */
+   CompletionStage<Void> run(Object... keys);
+
+   /**
+    * Remove all the indexes from the cache.
+    */
+   CompletionStage<Void> remove();
+
+   /**
+    * Remove all entities of particular class from the index of a cache.
+    */
+   CompletionStage<Void> remove(Class<?>... entities);
```

Comment:
This is ok for now, but it will need to also have an overloaded variant that takes type names as strings in the very near future (until 11 final).

Reply:
I am not sure it is doable for ISPN 11, Hibernate Search doesn't now about protobuf messages